src/v/storage/readers_cache.cc

```diff
@@ -54,16 +54,16 @@ readers_cache::put(std::unique_ptr<log_reader> reader) {
         return model::record_batch_reader(std::move(reader));
     }
     // check if requested reader belongs to one of the locked range
-    auto lock_it = std::find_if(
-      _locked_offset_ranges.begin(),
-      _locked_offset_ranges.end(),
-      [&reader](const offset_range& range) {
-          return reader->lease_range_base_offset() > range.second
-                 || reader->lease_range_end_offset() < range.first;
-      });
```

Comment:
Nice catch! The fix seems reasonable, though I'm curious how else this bug may manifest itself. Is the worst that can happen a minute-long deadlock? Or is there any way this can result in a segfault or segments being deleted incorrectly?

Reply:
one minute deadlock is how it manifests, it can not lead to any other more serious issue as read/write/truncate concurrency control is separate from readers cache.